include/dlaf/eigensolver/tridiag_solver/rot.h

```diff
@@ -134,24 +135,76 @@ struct GivensRotation {
   T s;         // sine
 };
 
+// Apply GivenRotations to tiles of the square sub-matrix identified by tile in range [i_begin, i_end).
+//
+// @param i_begin global tile index for both row and column identifying the start of the sub-matrix
+// @param i_end global tile index for both row and column identifying the end of the sub-matrix
+// @param rots GivenRotations to apply (element column indices of rotations are relative to the sub-matrix)
+// @param mat matrix where the sub-matrix is located
+//
+// @pre mat is not distributed
+// @pre memory layout of @p mat is column major.
+template <class T, Device D, class RotsSender>
+void applyGivensRotationsToMatrixColumns(const SizeType i_begin, const SizeType i_end, RotsSender&& rots,
+                                         Matrix<T, D>& mat) {
+  // Note:
+  // a column index may be paired to more than one other index, this may lead to a race
+  // condition if parallelized trivially. Current implementation is serial.
+
+  namespace ex = pika::execution::experimental;
+  namespace di = dlaf::internal;
+
+  const SizeType n = problemSize(i_begin, i_end, mat.distribution());
+  const SizeType nb = mat.distribution().blockSize().rows();
+
+  auto givens_rots_fn = [n, nb](const auto& rots, const auto& tiles, [[maybe_unused]] auto&&... ts) {
+    // Distribution of the merged subproblems
+    matrix::Distribution distr(LocalElementSize(n, n), TileElementSize(nb, nb));
+
+    for (const GivensRotation<T>& rot : rots) {
+      // Get the index of the tile that has column `rot.i` and the the index of the column within the tile.
+      const SizeType i_tile = distr.globalTileLinearIndex(GlobalElementIndex(0, rot.i));
+      const SizeType i_el = distr.tileElementFromGlobalElement<Coord::Col>(rot.i);
+      T* x = tiles[to_sizet(i_tile)].ptr(TileElementIndex(0, i_el));
+
+      // Get the index of the tile that has column `rot.j` and the the index of the column within the tile.
+      const SizeType j_tile = distr.globalTileLinearIndex(GlobalElementIndex(0, rot.j));
+      const SizeType j_el = distr.tileElementFromGlobalElement<Coord::Col>(rot.j);
+      T* y = tiles[to_sizet(j_tile)].ptr(TileElementIndex(0, j_el));
+
+      // Apply Givens rotations
+      if constexpr (D == Device::CPU) {
+        blas::rot(n, x, 1, y, 1, rot.c, rot.s);
```

Comment:
Add `SingleThreadedBlasScope` in this lambda.

Reply:
Done during the merge https://github.com/eth-cscs/DLA-Future/pull/843/commits/7332a5fe1dc6ae1c99f995fea84521391f740f05